### PR TITLE
Add hashcode and equals to ContextList

### PR DESCRIPTION
**Pull request by lun-li**

### CODE REVIEW

File: src/test/java/org/mondemand/tests/ClientTest.java [hunk 1]
```diff
@@ -1008,19 +1008,22 @@ public void testIncrementMap()
       client.increment(contexts, "key1");
       client.increment(contexts, "key2", 100l);
     }
+    ContextList contextsCopy = new ContextList();
+    contextsCopy.addContext(context);
```
Comment (nimaco) [inline src/test/java/org/mondemand/tests/ClientTest.java]: I'd suggest creating a new equivalent Context here as well.  As this stands, I think you are not covering much of Context.equals() with this test

**Comment by lun-li** [comment]
@nimaco Addressed your comment, please review again.

File: src/main/java/org/mondemand/Context.java [hunk 2]
```diff
@@ -86,9 +86,9 @@ public boolean equals(Object obj) {
     return true;
   }
 
-  public void addContext(Client client)
```
Comment (nimaco) [inline src/main/java/org/mondemand/Context.java]: This change technically breaks backwards compatibility, which would mean a major version bump

Comment (lun-li) [inline src/main/java/org/mondemand/Context.java]: @nimaco This code was added in the previous version and had never been used. Do we still want a major version bump?

**Comment by nimaco** [comment]
lgtm

**Pull request merged**